velox/vector/FlatVector.cpp

```diff
@@ -109,6 +109,11 @@ void FlatVector<StringView>::set(vector_size_t idx, StringView value) {
   if (BaseVector::rawNulls_) {
     BaseVector::setNull(idx, false);
   }
+  if (type()->kind() == TypeKind::VARCHAR) {
+    VELOX_CHECK(
```

Comment:
VELOX_CHECK_LE

> Value exceeds allowed Vector type size

This message is a bit vague. Would it be possible to clarify it a bit? Specifically, "Vector type" here is VARCHAR, no?

getVarcharLength seems expensive. Can we cache its value it in a member variable?

Reply:
Regarding `getVarcharLength`. The base type interface does not have a function that can return parameterized members. The parameters are already a member in the type (via `std::vector<TypeParameter> parameters_;`). But the TypePtr needs casting to the actual type to get access to it (in this case the `length()` function). 
I re-used the same approach that was used for DECIMAL.

Perhaps `length` isn't the right term and we could use a common `size()` function that is added to the types? For most of the primitive types it could the implementation size and for varchar/varbinary (and eventually char) it would be the definition size rather than implementation size but that could be leading to misunderstandings?